include/system_heap.h

```diff
@@ -0,0 +1,9 @@
+#ifndef SYSTEM_HEAP
+#define SYSTEM_HEAP
```

Comment:
I don't know what style MM goes by for include guards, but should these be
```suggestion
#ifndef SYSTEM_HEAP_H
#define SYSTEM_HEAP_H
```